File: 23 November 2022/Python/Maximum Sum LCM.py
#User function Template for python3
class Solution:
    def maxSumLCM (self, n):
        max_sum=0
        i=1
        while i**2<=n:
            if n%i==0:
                max_sum+=i
                if n//i!=i:
                    max_sum+=n//i
            i+=1
        return max_sum

#{ 
 # Driver Code Starts
#Initial Template for Python 3
if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        n = int(input())
        
        ob = Solution()
        print(ob.maxSumLCM(n))
# } Driver Code Ends

File: 23 November 2022/C++/Maximum Sum LCM.cpp
//{ Driver Code Starts
//Initial Template for C++
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
//User function Template for C++
class Solution {
  public:
    long long int maxSumLCM(int n) 
    { 
        long long int max_sum = 0;
        for (int i=1; i*i<=n; i++) 
        { 
            if (n%i == 0) 
            { 
                max_sum += i; 
                if (n/i != i) 
                    max_sum += (n/i); 
            } 
        } 
      
        return max_sum; 
    } 
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int  n;
        cin >> n ;
        Solution ob;
        cout<<ob.maxSumLCM(n)<<endl;
    }
    return 0;
}

// } Driver Code Ends